src/celengine/shadermanager.cpp

```diff
@@ -2501,7 +2474,7 @@ ShaderManager::buildFragmentShader(const ShaderProperties& props)
     {
         if (useSeparateDiffuse)
         {
-            for (unsigned k = 0; k < props.nLights - 1; k++)
+            for (unsigned k = 0; (k + 1) < props.nLights; ++k)
```

Comment:
why?

Reply:
`unsigned short` promotes to `int` in arithmetic operations, which then causes a signed vs unsigned comparison.

Looking at this further I'm not sure the statement generated here actually does anything?